### ISSUE TITLE
Add soft dynamic time warping metric

### ISSUE THREAD

**Issue by zaemyung** [subhadarship/kmeans_pytorch#20]
I wrote a pairwise distance metric that computes the soft dynamic time warping distance (https://github.com/subhadarship/kmeans_pytorch/pull/19).
The actual computation of the distance is done by [Maghoumi/pytorch-softdtw-cuda](https://github.com/Maghoumi/pytorch-softdtw-cuda).

Just to say a few words about the codes:
`SoftDTW` takes inputs with shape `(batch_size, seq_len, feature_dims)`.
As the kmeans in this repo deals only with univariate inputs, the `feaure_dims` is set to 1.
The tensors are broadcasted accordingly.
The code seems to work, but please correct them if I'm wrong!)

```python
def pairwise_soft_dtw(data1, data2, sdtw=None, device=torch.device('cpu')):
    if sdtw is None:
        raise ValueError('sdtw is None - initialize it with SoftDTW')

    # transfer to device
    data1, data2 = data1.to(device), data2.to(device)

    # (batch_size, seq_len, feature_dim=1)
    A = data1.unsqueeze(dim=2)

    # (cluster_size, seq_len, feature_dim=1)
    B = data2.unsqueeze(dim=2)

    distances = []
    for b in B:
        # (1, seq_len, 1)
        b = b.unsqueeze(dim=0)
        A, b = torch.broadcast_tensors(A, b)
        # (batch_size, 1)
        sdtw_distance = sdtw(b, A).view(-1, 1)
        distances.append(sdtw_distance)

    # (batch_size, cluster_size)
    dis = torch.cat(distances, dim=1)
    return dis
```

**Comment by subhadarship** [comment]
thanks for contributing!
